neo/public/javascripts/map/controllers.js

```diff
@@ -205,6 +169,88 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
 
       loadGeoJsonFiles(onEachFeature);
 
+      $scope.$on("leafletDirectiveMap.zoomend", function() {
```

Comment:
try to not move a large chunk of codes in the future. it makes review very hard.